gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java

```diff
@@ -187,9 +195,66 @@ public Optional<Dag<JobExecutionPlan>> validateAndHandleConcurrentExecution(Conf
    * @param allowConcurrentExecution
    * @return true if the {@link FlowSpec} allows concurrent executions or if no other instance of the flow is currently RUNNING.
    */
-  private boolean isExecutionPermitted(FlowStatusGenerator flowStatusGenerator, String flowGroup, String flowName,
-      boolean allowConcurrentExecution, long flowExecutionId) {
-    return allowConcurrentExecution || !flowStatusGenerator.isFlowRunning(flowName, flowGroup, flowExecutionId);
+  private boolean isExecutionPermitted(String flowGroup, String flowName, boolean allowConcurrentExecution)
+      throws IOException {
+    return allowConcurrentExecution || !isFlowRunning(flowGroup, flowName, dagManagementStateStore);
+  }
+
+  /**
+   * Returns true if any previous execution for the flow determined by the provided flowGroup, flowName is running.
+   * We ignore the execution that has the provided flowExecutionId. We also ignore the flows that are running beyond
+   * the job start deadline and flow finish deadline.
+   */
+  @VisibleForTesting
+  static boolean isFlowRunning(String flowGroup, String flowName, DagManagementStateStore dagManagementStateStore)
+      throws IOException {
+    List<FlowStatus> flowStatusList = dagManagementStateStore.getAllFlowStatusesForFlow(flowGroup, flowName);
```

Comment:
We need to ignore the flow execution ID of the flow currently being processed, since another host may have already emitted a status for it, thus causing split brain between the hosts.

Reply:
if another host has already emitted a status for it, then shouldnt this host know about it and thus take the right decision (of not submitting another flow)? am i missing something?